src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,202 @@
+import java.io.StreamCorruptedException;
+import java.time.LocalDateTime;
+
+public class Parser {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Parser(Storage storage, TaskList tasks, Ui ui) {
+        this.storage = storage;
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    // Takes a command and executes it accordingly
+    public void parse(String cmd) {
```

Comment:
maybe you can consider encapsulating the error handling in the markOrDelete() method to reduce repeated code and improve readability in this method